### PR TITLE
feat(broker): add port option

### LINKED ISSUE #40: Make port configurable
Needed for https://github.com/kotekan/kotekan/issues/255

### PR DESCRIPTION
**Pull request by nritsche**
This is in preparation to running an additional test on the CI servers of kotekan and comet that needs both programs. Since for kotekan CI tests run in parallel there needs to be an option to run comet with a random port.

- Make port configurable with -p.
- Choose and report a random port if set to 0.
- To reserve the chosen port, pass socket to server instead of `host,
  port`.

Closes #40

### CODE REVIEW

File: comet/broker.py [hunk 1]
```diff
@@ -570,27 +570,29 @@ def saveDataset(hash, ds, root):
...
             "file_lock_time": file_lock_time,
             "debug": debug,
             "recover": recover,
+            "port": port,
         }
 
         dumper = Dumper(data_dump_path, file_lock_time)
         self.debug = debug
         self.startup_time = datetime.datetime.utcnow()
+        self.port = None
```
Comment (anjakefala) [inline comet/broker.py]: so this is the actual port that ends up being connected to and `self.config['port']` is the configuration for the port.

File: comet/broker.py [hunk 2]
```diff
@@ -570,27 +570,29 @@ def saveDataset(hash, ds, root):
...
 
-    @staticmethod
```
Comment (anjakefala) [inline comet/broker.py]: why is it not a `@staticmethod` anymore?

File: comet/broker.py [hunk 3]
```diff
@@ -570,27 +570,29 @@ def saveDataset(hash, ds, root):
...
-    def _wait_and_register(startup_time, config):
+    def _wait_and_register(self):
         global dumper
-        sleep(1)
-        manager = Manager("localhost", DEFAULT_PORT)
+        while not self.port:
+            sleep(1)
+        manager = Manager("localhost", self.port)
```
Comment (anjakefala) [inline comet/broker.py]: What happens while `self.port = None`?

Comment (anjakefala) [reply comet/broker.py]: Ah, the above is a while loop. This does not execute until `self.port` gets a while.

**Comment by anjakefala** [comment]
I have a few questions, but everything looks consistent otherwise!

File: comet/broker.py (hunk 1 above)
Comment (nritsche) [reply comet/broker.py]: Yes. The the configuration can be `0` and then `self.port` would have the actual randomly chosen port.

File: comet/broker.py (hunk 3 above)
Comment (nritsche) [reply comet/broker.py]: Yeah, just to make sure. It's not like that will take longer than a second I hope.

File: comet/broker.py (hunk 2 above)
Comment (nritsche) [reply comet/broker.py]: I don't remember why it ever was.

**Comment by anjakefala** [comment]
All resolved!

File: comet/broker.py (hunk 2 above)
Comment (anjakefala) [reply comet/broker.py]: lol.

**Pull request merged**